Forward `run` task to `bgRun`

eed3si9n opened the pull request:
Fixes #3425

The `Compile`-specific `run` task was removed in #2936 in favor of `bgRun` but it stop short of delegating the `run` to `bgRun`.

eed3si9n committed:
Forward `run` task to `bgRun`

Fixes #3425

The `Compile`-specific `run` task was removed in #2936 in favor of `bgRun` but it stop short of delegating the `run` to `bgRun`.

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3477/head&deploy_to_bintray=false">this link.</A>

eed3si9n commented:
Travis failures thus far:

- actions / add-alias
- package / resources
- run / error
- run / fork-loader
- project / flatten
- source-dependencies / constants
- source-dependencies / implicit-search

eed3si9n committed:
Capture errors thrown inside `bgRun`

This captures non-fatal errors in `bgRun` background thread, so if needed we can re-throw it for `run`.

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3477/head&deploy_to_bintray=false">this link.</A>

eed3si9n committed:
Fix project/flatten

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3477/head&deploy_to_bintray=false">this link.</A>
